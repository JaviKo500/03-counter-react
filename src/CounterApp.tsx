interface Props {
  value: number;
}

export const CounterApp = ( { value } : Props ) => {
  return (
    <>
      <h1>Counter App</h1>
      <h2>{ value }</h2>
    </>
  )
}
