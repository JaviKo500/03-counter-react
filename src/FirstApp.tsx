const person = {
  name: 'Javiko500',
  age: 30,
  city: 'Santiago',
};

const currentYear = () => {
  return new Date().getFullYear();
}
let counter = 0;
const add = () => {
  return counter++;
}

export const FirstApp = () => {
  return (
    <>
      <h1>
        Hello
      </h1>
      <p>I'm { person.name }</p>
      <p>My age is { person.age } in { currentYear() }</p>
      <p>I live in { person.city }</p>
      <button onClick={ () => add() }>Click me { counter }</button>
      <br />
      <code>{ JSON.stringify(person) }</code>
    </>
  );
}
