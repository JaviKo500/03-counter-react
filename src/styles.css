html, body {
  background-color: #21232A;
  color: #fff;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 15px;
  font-size: 1.3rem;
  border-radius: 12px;
  background-color: #fff;
  margin-right: 5px;
}