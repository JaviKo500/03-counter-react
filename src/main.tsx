import { App } from './App';
import { StrictMode } from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
  <StrictMode>
    <App />
    <FirstApp title="Hello World" subtitle={'test subtitle'} showTitle />
  </StrictMode>
);