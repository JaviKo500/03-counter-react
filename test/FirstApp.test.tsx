import { render } from "@testing-library/react";
import { FirstApp } from '../src/FirstApp';
import React from "react";

describe('FirstApp.test', () => {
  test( 'check match with snapshot', () => {
    const title = 'Hello World';
    const { container } = render(<FirstApp title={title} showTitle/>);
    expect(container).toMatchSnapshot();
  });

  test( 'show title in h1', () => {
    const title = 'Hello World';
    const { getByText, container } = render(<FirstApp title={title} showTitle/>);
    const text = getByText(title)
    expect(text).toBeTruthy();

    const h1 = container.querySelector('h1');
    expect(h1?.innerHTML).toBe(title);
  });
}); 