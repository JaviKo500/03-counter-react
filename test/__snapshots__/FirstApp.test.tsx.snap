// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp.test check match with snapshot 1`] = `
<div>
  <h1
    style="display: block;"
  >
    Hello World
  </h1>
  <h3 />
  <p>
    I'm 
    Javiko500
  </p>
  <p>
    My age is 
    30
     in 
    2025
  </p>
  <p>
    I live in 
    Santiago
  </p>
  <button>
    Click me 
    0
  </button>
  <br />
</div>
`;
